core/src/main/java/org/apache/iceberg/ManifestEntriesTable.java

```diff
@@ -110,46 +111,56 @@ protected TableScan newRefinedScan(TableOperations ops, Table table, Schema sche
       ResidualEvaluator residuals = ResidualEvaluator.unpartitioned(filter);
 
       return CloseableIterable.transform(manifests, manifest ->
-          new ManifestReadTask(ops.io(), manifest, schema(), schemaString, specString, residuals,
+          ManifestEntriesReadTask.of(ops.io(), manifest, schema(), schemaString, specString, residuals,
               ops.current().specsById()));
     }
   }
 
-  static class ManifestReadTask extends BaseFileScanTask implements DataTask {
-    private final Schema schema;
-    private final Schema fileSchema;
-    private final FileIO io;
-    private final ManifestFile manifest;
-    private final Map<Integer, PartitionSpec> specsById;
+  @Value.Immutable
+  abstract static class ManifestEntriesReadTask extends BaseFileScanTask implements DataTask {
```

Comment:
I had to rename this class because the generated Immutable class was clashing with the `ManifestReadTask` from `DataFilesTable`